### PR TITLE
NavigatorNavigationBar fix this refernce issue

### PR DESCRIPTION
**Pull request by andyfenelon**
currently NavigatorNavigationBar is **broken**, this is due to a change on 0.32 where the scope of 'this' for the reset function has been change to an arrow function rather than referencing the class. 

see issue : https://github.com/facebook/react-native/issues/9610
this fix corrects this.

### CODE REVIEW

**Comment by facebook-github-bot-1** [comment]
By analyzing the blame information on this pull request, we identified @spicyj and @corbt to be potential reviewers.

**Comment by facebook-github-bot-1** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-5** [comment]
@andyfenelon updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9641/files/605ad17d2a20e08c7bdd0df42a2f502b814a1c36..2ba37fcc5b5cc3d9cb35a6cb9051bf24cf0a9446)

**Comment by facebook-github-bot-9** [comment]
@andyfenelon updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9641/files/2ba37fcc5b5cc3d9cb35a6cb9051bf24cf0a9446..2ac2813cc9c810b3e00fa5a10bd29c4a10ccf363)

**Comment by facebook-github-bot-9** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by satya164** [comment]
This doesn't look right. Arrow functions as class properties help to maintain the `this` variable, It looks weird that changing it to a class method will fix it.

**Comment by andyfenelon** [comment]
the alternative would be to bind it to the constructor and keep the arrow func ?

**Comment by facebook-github-bot-4** [comment]
@andyfenelon updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9641/files/7a75e4de9cce205cfe806d05eec5f407efc16e1c..a64ed4c90a557d27fdabe17a02015fe384b4523c)

**Comment by facebook-github-bot-6** [comment]
@andyfenelon updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9641/files/a64ed4c90a557d27fdabe17a02015fe384b4523c..c5b27cc9a2fc086e262a89ec79ba6090a31e4e32)

**Comment by andyfenelon** [comment]
updated to bind it too the constructor.

File: Libraries/CustomComponents/Navigator/NavigatorNavigationBar.js [hunk 1]
```diff
@@ -74,6 +74,11 @@ class NavigatorNavigationBar extends React.Component {
     navigationStyles: NavigatorNavigationBarStyles,
   };
 
+  constructor(props){
+    super(props);
+    this._reset = this._reset.bind(this)
```
Comment (corbt) [inline Libraries/CustomComponents/Navigator/NavigatorNavigationBar.js]: Arrow functions used as class properties are bound to `this` by default, so I don't think this should make any difference. Have you tested your app with/without this change and seen if it affects the behavior?

**Comment by andyfenelon** [comment]
seems fixed now. closing.

**Comment by samuelkraft** [comment]
@andyfenelon Fixed in master? Or does it work for you in 0.32?

**Comment by andyfenelon** [comment]
i grabbed the latest RN and used the code in the original issue and it seemed to work. so i just figured whatever the issue was had been patched?

**Pull request closed without merging**